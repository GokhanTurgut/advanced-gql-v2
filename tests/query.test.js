const gql = require("graphql-tag");

const createTestServer = require("./helper");

const FEED = gql`
  {
    feed {
      id
      message
      createdAt
      likes
      views
    }
  }
`;

const ME = gql`
  {
    me {
      id
      email
      verified
      role
      avatar
      createdAt
    }
  }
`;

describe("queries", () => {
  test("feed", async () => {
    const { query } = createTestServer({
      user: { id: 1 },
      models: {
        Post: {
          findMany: jest.fn(() => [
            {
              id: 1,
              message: "hello",
              createdAt: 12345839,
              likes: 20,
              views: 300,
            },
          ]),
        },
      },
    });

    const res = await query({ query: FEED });
    expect(res).toMatchSnapshot();
  });

  test("me", async () => {
    const { query } = createTestServer({
      user: {
        email: "test@test.com",
        password: "test",
        role: "ADMIN",
        verified: false,
        avatar: "http",
        createdAt: 1645708035209,
        id: "qi6haU8K48AjaqVvdoe7J",
      },
    });

    const res = await query({ query: ME });
    expect(res).toMatchSnapshot();
  });
});
