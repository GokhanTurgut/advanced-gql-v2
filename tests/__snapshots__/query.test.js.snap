// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`queries feed 1`] = `
Object {
  "data": Object {
    "feed": Array [
      Object {
        "createdAt": "12345839",
        "id": "1",
        "likes": 20,
        "message": "hello",
        "views": 300,
      },
    ],
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;

exports[`queries me 1`] = `
Object {
  "data": Object {
    "me": Object {
      "avatar": "http",
      "createdAt": "1645708035209",
      "email": "test@test.com",
      "id": "qi6haU8K48AjaqVvdoe7J",
      "role": "ADMIN",
      "verified": false,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": Object {
    "headers": Headers {
      Symbol(map): Object {},
    },
  },
}
`;
